File: techbuzz-springboot-java-api/src/main/java/com/sivalabs/techbuzz/entities/Link.java
package com.sivalabs.techbuzz.entities;

import javax.persistence.*;
import javax.validation.constraints.NotEmpty;
import java.util.Date;

@Entity
@Table(name="LINKS")
public class Link {

    @Id
    @Column(name = "id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotEmpty(message = "Title should not be empty")
    @Column(nullable = false)
    private String title;

    @NotEmpty(message = "URL should not be empty")
    @Column(nullable = false)
    private String url;

    @Temporal(TemporalType.TIMESTAMP)
    private Date createdOn = new Date();

    @ManyToOne
    @JoinColumn(name = "created_by", nullable = false)
    private User createdBy;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public Date getCreatedOn() {
        return createdOn;
    }

    public void setCreatedOn(Date createdOn) {
        this.createdOn = createdOn;
    }

    public User getCreatedBy() {
        return createdBy;
    }

    public void setCreatedBy(User createdBy) {
        this.createdBy = createdBy;
    }
}

File: src/main/kotlin/com/sivalabs/techcorner/entities/Link.kt
package com.sivalabs.techcorner.entities

import java.time.LocalDateTime
import javax.persistence.*
import javax.persistence.JoinColumn
import javax.persistence.JoinTable
import javax.persistence.ManyToMany

@Entity
@Table(name = "links")
class Link {
    @Id
    @SequenceGenerator(name = "link_generator", sequenceName = "link_sequence", initialValue = 10)
    @GeneratedValue(generator = "link_generator")
    var id: Long? = null

    @Column(nullable = false)
    var title: String = ""

    @Column(nullable = false)
    var url: String = ""

    @ManyToMany(cascade = [(CascadeType.MERGE)])
    @JoinTable(name = "link_tag", joinColumns = [JoinColumn(name = "link_id", referencedColumnName = "ID")],
            inverseJoinColumns = [JoinColumn(name = "tag_id", referencedColumnName = "ID")])
    var tags: MutableList<Tag> = mutableListOf()

    @JoinColumn(nullable = false, name = "created_by")
    @ManyToOne()
    var createdBy: User = User()

    @Column(nullable = false)
    var createdOn: java.time.LocalDateTime = LocalDateTime.now()
}